components/channel_view/channel_view.tsx

```diff
@@ -16,33 +17,41 @@ import TutorialView from 'components/tutorial';
 import {clearMarks, mark, measure, trackEvent} from 'actions/telemetry_actions.jsx';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 
-export default class ChannelView extends React.PureComponent {
-    static propTypes = {
-        channelId: PropTypes.string.isRequired,
-        deactivatedChannel: PropTypes.bool.isRequired,
-        channelRolesLoading: PropTypes.bool.isRequired,
-        match: PropTypes.shape({
-            url: PropTypes.string.isRequired,
-            params: PropTypes.shape({
-                postid: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-        showTutorial: PropTypes.bool.isRequired,
-        showNextSteps: PropTypes.bool.isRequired,
-        showNextStepsTips: PropTypes.bool.isRequired,
-        isOnboardingHidden: PropTypes.bool.isRequired,
-        showNextStepsEphemeral: PropTypes.bool.isRequired,
-        channelIsArchived: PropTypes.bool.isRequired,
-        viewArchivedChannels: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            goToLastViewedChannel: PropTypes.func.isRequired,
-            setShowNextStepsView: PropTypes.func.isRequired,
-            getProfiles: PropTypes.func.isRequired,
-        }),
-        isCloud: PropTypes.bool.isRequired,
-    };
-
-    static createDeferredPostView = () => {
+type Props = {
+    channelId: string;
+    deactivatedChannel: boolean;
+    channelRolesLoading: boolean;
+    match: {
+        url: string,
+        params: {
+            postid?: string,
+        }
+    },
+    showTutorial: boolean;
+    showNextSteps: boolean;
+    showNextStepsTips: boolean;
+    isOnboardingHidden: boolean;
+    showNextStepsEphemeral: boolean;
+    channelIsArchived: boolean;
+    viewArchivedChannels: boolean;
+    isCloud: boolean;
+    actions: {
+        goToLastViewedChannel: () => Promise<{data: boolean}>;
+        setShowNextStepsView: (show: boolean) => Action;
+        getProfiles: (page?: number, perPage?: number, options?: any) => ActionFunc;
+    },
+};
+
+type State = {
+    channelId: string;
+    url: string;
+    focusedPostId?: string;
+    deferredPostView: any;
```

Comment:
This any too

Reply:
@ethervoid Hopefully this isn't a bother but I'm not sure how to tackle this one? I've tried changing this to `typeof DeferredRenderWrapper & hoistNonReactStatics.NonReactStatics<any, {}>` but then it creates this error on line 104:

`Type 'typeof DeferredRenderWrapper & NonReactStatics<any, {}>' is not assignable to type '((WrappedComponent: any, PreRenderComponent?: any) => typeof DeferredRenderWrapper & NonReactStatics<any, {}>) & NonReactStatics<any, {}>'.
  Type 'typeof DeferredRenderWrapper & NonReactStatics<any, {}>' is not assignable to type '(WrappedComponent: any, PreRenderComponent?: any) => typeof DeferredRenderWrapper & NonReactStatics<any, {}>'.
    Type 'typeof DeferredRenderWrapper & NonReactStatics<any, {}>' provides no match for the signature '(WrappedComponent: any, PreRenderComponent?: any): typeof DeferredRenderWrapper & NonReactStatics<any, {}>'.`

and this error on line 248:

`'DeferredPostView' cannot be used as a JSX component.
  Its return type 'typeof DeferredRenderWrapper & NonReactStatics<any, {}>' is not a valid JSX element.
    Type 'typeof DeferredRenderWrapper & NonReactStatics<any, {}>' is missing the following properties from type 'Element': type, props, key``

I was also trying to search if it was a custom type in the mattermost-redux library but there is not one that seems to fit this.

Reply:
Ok, we can leave as any for now :)